### PR TITLE
[SW-816, SW-828] Display a low balance message when sending an approved multisig transaction.

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

File: src/popup/utils/helper.js
Comment (pstachel) [inline src/popup/utils/helper.js:1]: Code for helper.js borrowed from https://github.com/aeternity/superhero-wallet/pull/1989/files#diff-08d84b32fbf152c4407ebc44276a4df0362fd3c88bc8e226d6a33c9c1d15e6d7R186-R189

File: src/popup/pages/MultisigProposalDetails.vue [hunk 1]
```diff
@@ -439,7 +443,11 @@ export default defineComponent({
 
         proposalCompleted.value = true;
       } catch (error) {
-        handleUnknownError(error);
+        if (isInsufficientBalanceError(error)) {
+          openModal({ name: MODAL_DEFAULT, icon: 'alert', msg: root.$t('modals.vaultLowBalance.msg') });
+        } else {
+          handleUnknownError(error);
```
Comment (subhod-i) [inline src/popup/pages/MultisigProposalDetails.vue:449]: We can notify the user if there is any other error by using a modal.
I am okay with the current approach too. Up to you @pstachel

**Review by subhod-i — APPROVED** [review]

File: src/popup/pages/MultisigProposalDetails.vue (hunk 1 above)
Comment (pstachel) [reply src/popup/pages/MultisigProposalDetails.vue:449]: You're right, I'll adjust the code

File: src/popup/pages/MultisigProposalDetails.vue [hunk 2]
```diff
@@ -595,7 +602,7 @@ export default defineComponent({
   }
 
   .bottom-buttons {
-    padding: 24px;
+    padding: var(--screen-padding-x);
```
Comment (CedrikNikita) [inline src/popup/pages/MultisigProposalDetails.vue:605]: The variable value and the previous value of the padding are different. Are you sure with this changes?

Comment (pstachel) [reply src/popup/pages/MultisigProposalDetails.vue:605]: That's what SW-828 was about - to align the padding of the buttons with the view

File: src/popup/components/Modals/Default.vue [hunk 3]
```diff
@@ -19,7 +19,14 @@
...
 
-    <slot name="msg">
+    <TemplateRenderer
+      v-if="!$slots.msg"
+      :str="msg"
+    />
+    <slot
+      v-else
+      name="msg"
+    >
```
Comment (peronczyk) [inline src/popup/components/Modals/Default.vue:29]: No need to use v-if here. Just do this:
```
<slot name="msg">
  <TemplateRenderer ... />
</slot>
```

File: src/popup/locales/en.json [hunk 4]
```diff
@@ -119,6 +119,10 @@
     "insufficient-balance": {
       "msg": "You do not have enough balance for this transaction."
     },
+    "vaultLowBalance": {
+      "title": "Insufficient vault balance",
+      "msg": "Multisig vault balance is not enough to pay for the transaction fee. <br /><br />Please top up multisig vault."
```
Comment (peronczyk) [inline src/popup/locales/en.json:124]: Please remove the space before BRs :)

File: src/popup/utils/helper.js [hunk 5]
```diff
@@ -181,3 +182,13 @@ export const isValidURL = (url) => {
...
+
+const hasValidationKey = (error, expectedKey) => (
+  error.validation
+  && error.validation.some(({ key }) => expectedKey === key)
+);
+
+export const isInsufficientBalanceError = (error) => (
+  error instanceof InvalidTxError
+  && hasValidationKey(error, 'InsufficientBalance')
+);
```
Comment (peronczyk) [inline src/popup/utils/helper.js:194]: Please avoid adding new utils to non-TS files. There is pending PR for moving all stuff from this file into the utils.ts file.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/utils/utils.ts [hunk 6]
```diff
@@ -506,3 +507,13 @@ export function isTxOfASupportedType(encodedTx: string, isTxBase64 = false) {
...
+
+const hasValidationKey = (error: any, expectedKey: string) => (
+  error.validation
+  && error.validation.some(({ key }: any) => expectedKey === key)
+);
+
+export const isInsufficientBalanceError = (error: any) => (
+  error instanceof InvalidTxError
+  && hasValidationKey(error, 'InsufficientBalance')
+);
```
Comment (peronczyk) [inline src/popup/utils/utils.ts:519]: Please use regular functions instead of arrows same as all other functions here.

**Review by peronczyk — APPROVED** [review]

**Pull request merged**